tls/s2n_client_hello.c

```diff
@@ -652,10 +652,24 @@ int s2n_client_hello_send(struct s2n_connection *conn)
     return S2N_SUCCESS;
 }
 
-/* See http://www-archive.mozilla.org/projects/security/pki/nss/ssl/draft02.html 2.5 */
+/*
+ * s2n-tls does NOT support SSLv2. However, it does support SSLv2 ClientHellos.
+ * Clients may send SSLv2 ClientHellos advertising higher protocol versions for
+ * backwards compatibility reasons. See https://tools.ietf.org/rfc/rfc2246 Appendix E.
+ *
+ * In this case, conn->client_hello_version will be SSLv2, but conn->client_protocol_version
+ * will likely be higher.
+ *
+ * See http://www-archive.mozilla.org/projects/security/pki/nss/ssl/draft02.html Section 2.5
+ * for a description of the expected SSLv2 format.
+ * Alternatively, the TLS1.0 RFC includes a more modern description of the format:
+ * https://tools.ietf.org/rfc/rfc2246 Appendix E.1
+ */
 int s2n_sslv2_client_hello_recv(struct s2n_connection *conn)
 {
     struct s2n_client_hello *client_hello = &conn->client_hello;
+    client_hello->sslv2 = true;
```

Comment:
`s2n_sslv2_client_hello_recv` appears to only be called if and only if `conn->client_hello_version == S2N_SSLv2`. I realize the change that depends on `client_hello->sslv2` is not in this PR but I'm not understanding the need for two separate pieces of state here.

Reply:
client_hello_version is on the connection, but we vend multiple APIs that only take the client hello. Those include APIs to retrieve the raw ClientHello (which is different for sslv2) and the raw cipher suites (also different for sslv2). The client hello needs to know how its fields are formatted.